src/main/java/bio/terra/service/filedata/flight/delete/DeleteFileAzurePrimaryDataStep.java

```diff
@@ -0,0 +1,38 @@
+package bio.terra.service.filedata.flight.delete;
+
+import bio.terra.service.filedata.azure.blobstore.AzureBlobStorePdao;
+import bio.terra.service.filedata.flight.FileMapKeys;
+import bio.terra.service.filedata.google.firestore.FireStoreFile;
+import bio.terra.stairway.FlightContext;
+import bio.terra.stairway.FlightMap;
+import bio.terra.stairway.Step;
+import bio.terra.stairway.StepResult;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class DeleteFileAzurePrimaryDataStep implements Step {
+  private static final Logger logger =
+      LoggerFactory.getLogger(DeleteFileAzurePrimaryDataStep.class);
+
+  private final AzureBlobStorePdao azureBlobStorePdao;
+
+  public DeleteFileAzurePrimaryDataStep(AzureBlobStorePdao azureBlobStorePdao) {
+    this.azureBlobStorePdao = azureBlobStorePdao;
+  }
+
+  @Override
+  public StepResult doStep(FlightContext context) {
+    FlightMap workingMap = context.getWorkingMap();
+    FireStoreFile fireStoreFile = workingMap.get(FileMapKeys.FIRESTORE_FILE, FireStoreFile.class);
+    if (fireStoreFile != null) {
+      azureBlobStorePdao.deleteFile(fireStoreFile);
```

Comment:
why could this be null?

Reply:
That happens if you're trying to delete that can't be found in Firestore.  We don't fail, we noop

Reply:
ah gotcha. Probably not worth storing the file wrapped in an Optional in the workingMap, since type erasure is a thing :(

Reply:
Yeah...and I'm honestly not sure how that would work with the deserializer